pkg/validations/createcluster/createcluster.go

```diff
@@ -0,0 +1,90 @@
+package createcluster
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/gitops/flux"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/providers/validator"
+	"github.com/aws/eks-anywhere/pkg/validations"
+)
+
+type validationManager struct {
+	clusterSpec       *cluster.Spec
+	provider          providers.Provider
+	gitOpsFlux        *flux.Flux
+	createValidations Validator
+	dockerExec        validations.DockerExecutable
+}
+
+type Validator interface {
+	PreflightValidations(ctx context.Context) error
```

Comment:
Is it very difficult to have a `GetValidations() []validations.Validation` method?
I'm assuming the other object is creating a slice of validations and executing them. It should be as easy as splitting those two steps, right?